argoverse/map_representation/map_api_v2.py

```diff
@@ -0,0 +1,462 @@
+
+        return das
+        
+    def get_scenario_vector_drivable_areas(self) -> List[np.ndarray]:
+        """Fetch a list of polygons, whose union represents the drivable area for the log/scenario.
+
+        Note: this function provides drivable areas in vector, not raster, format).
+        """
+        return self._drivable_areas_list
+
+
+    def get_lane_segment_successor_ids(self, lane_segment_id: int) -> Optional[List[int]]:
+        """Get lane id for the lane sucessor of the specified lane_segment_id.
+
+        Args:
+            lane_segment_id: unique identifier for a lane segment within a log scenario map (within a single city).
+
+        Returns:
+            successor_ids: list of integers, representing lane segment IDs of successors
+        """
+        successor_ids = self._lane_segments_dict[lane_segment_id].successors
+        return successor_ids
+
+    def get_lane_segment_left_neighbor_id(self, lane_segment_id: int) -> Optional[int]:
+        """Get id of lane segment that is the left neighbor (if any exists) to the query lane segment id.
+
+        Args:
+            lane_segment_id: unique identifier for a lane segment within a log scenario map (within a single city).
+
+        Returns:
+            integer representing id of left neighbor to the query lane segment id, or None if no such neighbor exists.
+        """
+        return self._lane_segments_dict[lane_segment_id].l_neighbor_id
+
+    def get_lane_segment_right_neighbor_id(self, lane_segment_id: int) -> Optional[int]:
+        """Get id of lane segment that is the right neighbor (if any exists) to the query lane segment id.
+
+        Args:
+            lane_segment_id: unique identifier for a lane segment within a log scenario map (within a single city).
+
+        Returns:
+            integer representing id of right neighbor to the query lane segment id, or None if no such neighbor exists.
+        """
+        return self._lane_segments_dict[lane_segment_id].r_neighbor_id
+
+    def get_scenario_lane_segment_ids(self) -> List[int]:
+        """Get ids of all lane segments that are local to this log/scenario (according to l-infinity norm).
+
+        Returns:
+            list containing ids of local lane segments
+        """
+        return list(self._lane_segments_dict.keys())
+
+    def get_lane_segment_centerline(self, lane_segment_id: int, city_name: str) -> np.ndarray:
+        """We return an inferred 3D centerline for any particular lane segment by forming a ladder of left and right waypoints.
+
+        Args:
+            lane_segment_id: unique identifier for a lane segment within a log scenario map (within a single city).
+
+        Returns:
+            lane_centerline: Numpy array of shape (N,3)
+        """
+        left_ln_bnd = self._lane_segments_dict[lane_segment_id].left_ln_bnd
+        right_ln_bnd = self._lane_segments_dict[lane_segment_id].right_ln_bnd
+
+        lane_centerline = ""  # TODO: add 3d linear interpolation fn
```

Comment:
I'm wondering if having lane boundaries with differing numbers of points might throw this function off. If so, we'll probably have to add a check to ensure that all lane boundaries are interpolated to have the same number of points.

Reply:
no worries there since that is how we have it implemented for the 2d case, and I'll make it the same for the 3d case -- an equal number of points on the left and right (ladder), and find the midpoint of each